src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java

```diff
@@ -1730,17 +1730,207 @@ public void testDeleteProviderConfigWithNotFoundError() throws Exception {
   }
 
   @Test
-  public void testTenantAwareDeleteProviderConfig() throws Exception {
+  public void testTenantAwareDeleteOidcProviderConfig() throws Exception {
     TestResponseInterceptor interceptor = initializeAppForTenantAwareUserManagement(
         "TENANT_ID",
         "{}");
     TenantAwareFirebaseAuth tenantAwareAuth =
         FirebaseAuth.getInstance().getTenantManager().getAuthForTenant("TENANT_ID");
 
-    tenantAwareAuth.deleteOidcProviderConfig("PROVIDER_ID");
+    tenantAwareAuth.deleteOidcProviderConfig("oidc.provider-id");
 
     checkRequestHeaders(interceptor);
-    checkUrl(interceptor, "DELETE", TENANTS_BASE_URL + "/TENANT_ID/oauthIdpConfigs/PROVIDER_ID");
+    String expectedUrl = TENANTS_BASE_URL + "/TENANT_ID/oauthIdpConfigs/oidc.provider-id";
+    checkUrl(interceptor, "DELETE", expectedUrl);
+  }
+
+  @Test
+  public void testCreateSamlProvider() throws Exception {
+    TestResponseInterceptor interceptor = initializeAppForUserManagement(
+        TestUtils.loadResource("saml.json"));
+    // TODO(micahstairs): Add 'signRequest' to the create request once that field is added to
+    // SamlProviderConfig.
+    SamlProviderConfig.CreateRequest createRequest =
+        new SamlProviderConfig.CreateRequest()
+          .setProviderId("saml.provider-id")
+          .setDisplayName("DISPLAY_NAME")
+          .setEnabled(true)
+          .setIdpEntityId("IDP_ENTITY_ID")
+          .setSsoUrl("https://example.com/login")
+          .addX509Certificate("certificate1")
+          .addX509Certificate("certificate2")
+          .setRpEntityId("RP_ENTITY_ID")
+          .setCallbackUrl("https://projectId.firebaseapp.com/__/auth/handler");
+
+    SamlProviderConfig config = FirebaseAuth.getInstance().createSamlProviderConfig(createRequest);
+
+    checkSamlProviderConfig(config, "saml.provider-id");
+    checkRequestHeaders(interceptor);
+    checkUrl(interceptor, "POST", PROJECT_BASE_URL + "/inboundSamlConfigs");
+    GenericUrl url = interceptor.getResponse().getRequest().getUrl();
+    assertEquals("saml.provider-id", url.getFirst("inboundSamlConfigId"));
+
+    GenericJson parsed = parseRequestContent(interceptor);
+    assertEquals("DISPLAY_NAME", parsed.get("displayName"));
+    assertTrue((boolean) parsed.get("enabled"));
+    Map<String, Object> idpConfig = (Map<String, Object>) parsed.get("idpConfig");
```

Comment:
Just a hunch. See if you can use `GenericJson` in place of `Map<String, Object>`. If it works, it would clean things up a bit.

Reply:
If I try to cast it to `GenericJson` I get the following error:

```java.lang.ClassCastException: com.google.api.client.util.ArrayMap cannot be cast to com.google.api.client.json.GenericJson```

Reply:
Thanks for checking.